File: leetcode/exercises/1.1.1.easy.contains_duplicate_1.go
// -----------------------
// Created     : 30/10/2025
// Last Edited : 30/10/2025
// Topics      : 
// Big O       :
// Problem Id  : 217
// Source      :
// Notes       :
// -----------------------

package main
import "fmt"

func containsDuplicate(nums []int) bool {
    m := map[int]int{}
    for _, v := range nums {
        m[v] = m[v] + 1
        if (m[v] > 1) {
            return true
        }
    }
    return false
}

func main() {
    fmt.Println(containsDuplicate([]int{1,2,4,8,5,10}))
}

File: leetcode/exercises/1.1.1.easy.contains_duplicate_1.py
# -----------------------
# Created     : 22/10/2025
# Last Edited : 22/10/2025
# Topics      : 
# Big O       :
# Problem Id  : 217
# References  :
# Notes       :
# -----------------------

class Solution:
	def containsDuplicate(self, nums: list[int]) -> bool:
		a = set()
		for i in nums:
			if i not in a:
				a.add(i)
			else:
				return True
		return False

sol = Solution()
print()
print(sol.containsDuplicate([2, 1, 10, 12, 1]))
